### PR TITLE
feat: standalone visual context CSS generation

### PR DESCRIPTION
**Pull request by amanabiy**
### *Issue #, if available:*
This change allows new visual contexts to be compiled to separate CSS files that are only included in the bundle when the component is imported, making context addition scalable without growing the base bundle.

### *Description of changes:*
Adds a destination field to the Context interface. When set, the context is excluded from the base styles and compiled to its own CSS file at the specified path. The component that uses the context imports this file directly.

```typescript
// In the context definition:
{
  id: 'navigation-bar',
  selector: '.awsui-context-navigation-bar',
  destination: 'visual-contexts/navigation-bar.scoped.css',
  defaultMode: 'dark',
  tokens: { ... }
}

// In the component:
import '../visual-contexts/navigation-bar.scoped.css';
```

### Tested with
- [components PR #4477](https://github.com/cloudscape-design/components/pull/4477) ([last commit](https://github.com/cloudscape-design/components/pull/4477/changes/fc15fd44b8608b8f7b52c9b552d9a8e5c575232f)) consuming directly from this branch
- Generated file verified: correct token overrides, :not(#\9) specificity, excluded from base styles, existing contexts unchanged

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/build/internal.ts [hunk 1]
```diff
@@ -13,6 +14,11 @@ export { buildStyles, InlineStylesheet, BuildStylesOptions };
 
 export type Tasks = 'preset' | 'design-tokens';
 
+function stripStandaloneContexts(theme: Theme): Theme {
+  const contexts = Object.fromEntries(Object.entries(theme.contexts).filter(([_, ctx]) => !ctx.destination));
```
Comment (amanabiy) [inline src/build/internal.ts]: [nit: Not blocker]: I know I wrote it :). can we change the variable name from ctx to context?

**Review by amanabiy — COMMENTED** [review]
Looks good to me!

File: src/shared/declaration/index.ts [hunk 2]
```diff
@@ -97,13 +116,62 @@ export function createBuildDeclarations(
...
+      });
+
+    const stylesheet = buildStylesheet(contextThemes, propertiesMap, (selector) => selector, usedTokens);
+    const transformed = new MinimalTransformer().transform(stylesheet);
+    transformed.retainRulesMatching(contextThemes[0].contexts[contextId].selector);
+
+    // Standalone contexts must be wrapped with the same layer as base theme.
+    const css = transformed.toString('awsui-base-theme');
+    if (css.trim()) {
+      result[destination] = css;
```
Comment (amanabiy) [inline src/shared/declaration/index.ts:178]: If two different contexts use the same destination, this assignment replaces the output from the first context, so the generated file retains only the latest one. Can we handle this edge case by ensuring that destinations are unique across different context IDs and cover this behavior with a test?

Comment (pan-kot) [reply src/shared/declaration/index.ts:178]: Good point, addressed 👍

File: src/build/internal.ts (hunk 1 above)
Comment (pan-kot) [reply src/build/internal.ts]: Sure, updated

**Review by pan-kot — APPROVED** [review]

**Review by amanabiy — COMMENTED** [review]
Looks good to me now!

File: src/shared/theme/interfaces.ts [hunk 3]
```diff
@@ -27,6 +27,8 @@ export interface Context {
   selector: string;
   tokens: Record<string, Assignment>;
   defaultMode?: keyof Mode['states'];
+  /** If set, this context is excluded from base-component styles and output to this path instead. */
+  destination?: string;
```
Comment (jperals) [inline src/shared/theme/interfaces.ts:31]: Can we enable this behavior automatically to a generated path namespaced to the component instead of making this opt-in? Are there any risks or tradeoffs to use it for every component right away?

Comment (pan-kot) [reply src/shared/theme/interfaces.ts:31]: We cannot apply this to most of the existing contexts (can probably use this for compact table). The reason is that we have consumers that use contexts directly by attaching `awsui-context-*` classnames. If the contexts are moved - they will only be loaded to the app if the corresponding components are imported.

Comment (jperals) [reply src/shared/theme/interfaces.ts:31]: So this can only be used for components for which we do not plan to expose the visual context classnames?

Comment (pan-kot) [reply src/shared/theme/interfaces.ts:31]: This should be used for all new visual contexts and can be used for existing contexts that are considered safe, such as compact-table, unless we find confirmations of teams using this context explicitly, too.

Comment (jperals) [reply src/shared/theme/interfaces.ts:31]: So to make sure I understand, this is safe for customers to do:
```
.awsui-context-x {
   (...)
}
```

and this is not because it won't work until the component that loads the styles is loaded?
```
<div class="awsui-content-x"/>
```

In that case, looks like we need to document this in our API contract.

**Review by jperals — APPROVED** [review]

File: src/shared/theme/interfaces.ts (hunk 3 above)
Comment (pan-kot) [reply src/shared/theme/interfaces.ts:31]: Currently, the docs around direct use of contexts do not exist - but we will document that eventually. For contexts that are not merged into base component styles it will be required to either import the component that imports the context, or import the context css directly.
